cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/DatasetNamespace.java

```diff
@@ -16,28 +16,51 @@
 
 package co.cask.cdap.data2.dataset2;
 
+import co.cask.cdap.proto.Id;
+
 import javax.annotation.Nullable;
 
 /**
- * Performs namespacing for data set names.
+ * Performs namespacing for datasets.
  */
 public interface DatasetNamespace {
+
   /**
-   * @param name name of the dataset
-   * @return namespaced name of the dataset
+   * @param datasetInstanceName name of the dataset instance to be namespaced
+   * @return namespaced {@link Id.DatasetInstance} of the dataset
    */
-  String namespace(String name);
+  Id.DatasetInstance namespace(String datasetInstanceName);
 
   /**
-   * @param name namespaced name of the dataset
-   * @return original name of the dataset or null if name is not within this namespace
+   * @param datasetInstanceId {@link Id.DatasetInstance} for the dataset instance to be namespaced
```

Comment:
Oh. "namespace" here means something different than the namespace concept. This is very confusing...